docs/topics/testing/advanced.txt

```diff
@@ -600,9 +600,13 @@ utility methods in the ``django.test.utils`` module.
 
 .. function:: setup_test_environment()
 
-    Performs any global pre-test setup, such as the installing the
-    instrumentation of the template rendering system and setting up
-    the dummy email outbox.
+    Performs global pre-test setup, like installing instrumentation for the
+    template rendering system and setting up the dummy email outbox.
+
+    This function also saves the values of settings that it changes so that
```

Comment:
This is probably such a rare problem, but I agree it would be nice to prevent it just in case. How about instead detecting it and raising an error about it. What do you think? That would avoid some documentation that likely doesn't affect many people.